### PR TITLE
add stats

### PR DESCRIPTION
**Pull request by greenbigfrog**

### CODE REVIEW

File: src/discordtipbot/discordbot.cr [hunk 1]
```diff
@@ -517,6 +519,12 @@ class DiscordBot
     reply(msg, "Bot has been running for #{Time.now - START_TIME}")
   end
 
+  def stats(msg : Discord::Message)
+    guilds = @cache.guilds.size
+    users = @cache.users.size
```
Comment (z64) [inline src/discordtipbot/discordbot.cr]: This only reflects the number of users you happen to have cached in memory (typically only online users). A more useful stat might be `@guilds.values.map(&.member_count).sum` and describing it as "approximate users". Sure this does not account for the same user in multiple guilds, but it is very rare that a user is in more than one of the guilds your bot is in in the bigger picture.

Comment (greenbigfrog) [reply src/discordtipbot/discordbot.cr]: Hm.
Does `@cache.guilds` always contain all guilds? Else I could maybe make it `~ X guilds (cached: Y) and ~ A users (cached: B)

Comment (z64) [reply src/discordtipbot/discordbot.cr]: All guilds the shard is connected to is always received after `READY`, but `GUILD_CREATE` only contains online user presences at the time of startup.

Comment (greenbigfrog) [reply src/discordtipbot/discordbot.cr]: After 100 members I assume. Else the count would be currently off.

Comment (greenbigfrog) [reply src/discordtipbot/discordbot.cr]:
```
  def stats(msg : Discord::Message)
    guilds = @cache.guilds.size
    cached_users = @cache.users.size
    users = @cache.guilds.values.map(&.member_count).sum

    reply(msg, "The bot is in #{guilds} Guilds and sees #{users} users (of which #{cached_users} users are guaranteed unique)")
  end```

Comment (greenbigfrog) [reply src/discordtipbot/discordbot.cr]: @z64 does this look good?

Comment (z64) [reply src/discordtipbot/discordbot.cr]: Up to you how you want to handle it. `@cache.users.size` ultimately can't be relied on (the error will only grow with the number of guilds your on, with more error and less reliability than summing `member_count`)

**Comment by greenbigfrog** [comment]
CI fails because shard.lock hasn't been updated yet

**Pull request merged**